src/java/org/apache/cassandra/db/compaction/UnifiedCompactionContainer.java

```diff
@@ -0,0 +1,331 @@
+import org.apache.cassandra.notifications.INotification;
+import org.apache.cassandra.repair.consistent.admin.CleanupSummary;
+import org.apache.cassandra.schema.CompactionParams;
+
+public class UnifiedCompactionContainer implements CompactionStrategyContainer
+{
+    private final CompactionParams params;
+    private final CompactionParams metadataParams;
+    private final UnifiedCompactionStrategy strategy;
+
+    AtomicBoolean enabled;
+
+    UnifiedCompactionContainer(CompactionStrategyFactory factory,
+                               BackgroundCompactions backgroundCompactions,
+                               CompactionParams params,
+                               CompactionParams metadataParams,
+                               boolean enabled)
+    {
+        ColumnFamilyStore cfs = factory.getCfs();
+        this.params = params;
+        this.metadataParams = metadataParams;
+        this.strategy = new UnifiedCompactionStrategy(factory, backgroundCompactions, params.options());
+        this.enabled = new AtomicBoolean(enabled);
+
+        factory.getCompactionLogger().strategyCreated(this.strategy);
+
+        if (this.strategy.getOptions().isLogAll())
+            factory.getCompactionLogger().enable();
+        else
+            factory.getCompactionLogger().disable();
+
+        startup();
+    }
+
+    @Override
+    public void enable()
+    {
+        this.enabled.set(true);
+    }
+
+    @Override
+    public void disable()
+    {
+        this.enabled.set(false);
+    }
+
+    @Override
+    public boolean isEnabled()
+    {
+        return enabled.get() && strategy.isActive;
+    }
+
+    @Override
+    public boolean isActive()
+    {
+        return strategy.isActive;
+    }
+
+    @Override
+    public void reload(@Nonnull CompactionStrategyContainer previous, CompactionParams compactionParams, ReloadReason reason)
+    {
+        throw new UnsupportedOperationException("Not supported, create a new container");
+    }
+
+    @Override
+    public CompactionParams getCompactionParams()
+    {
+        return params;
+    }
+
+    @Override
+    public CompactionParams getMetadataCompactionParams()
+    {
+        return metadataParams;
+    }
+
+    @Override
+    public List<List<CompactionStrategy>> getStrategies()
+    {
+        return ImmutableList.of(ImmutableList.of(strategy));
+    }
+
+    @Override
+    public List<AbstractStrategyHolder.GroupedSSTableContainer> groupSSTables(Iterable<SSTableReader> sstables)
+    {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public AbstractCompactionTask findUpgradeSSTableTask()
+    {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    @Override
+    public int getUnleveledSSTables()
+    {
+        return 0;
+    }
+
+    @Override
+    public CleanupSummary releaseRepairData(Collection<UUID> sessions)
+    {
+        throw new UnsupportedOperationException("Not yet implemented");
```

Comment:
Implemented in 53df0b8f02d66a52b026e666f29faf68839e0b6f. *Note:* `CleanupTask` has the Apache license because I only extracted it - the code was already present in the project.